client/blocks/comment-detail/comment-detail-author.jsx

```diff
@@ -164,4 +201,12 @@ export class CommentDetailAuthor extends Component {
 	}
 }
 
-export default localize( CommentDetailAuthor );
+const mapStateToProps = ( state, ownProps ) => {
+	const settings = getSiteSettings( state, ownProps.siteId );
+	return {
+		dateFormat: get( settings, 'date_format' ),
+		timeFormat: get( settings, 'time_format' ),
```

Comment:
A new utility function was my first idea, but it's out of scope here.
Anyway, as far as I know (and as far as a project-wide search knows), the date/time formats are only used frontend side.